### ISSUE TITLE
Bug:fail to encode domain(test_a.abc.com)

### ISSUE THREAD

**Issue by ZhaoQi99** [kjd/idna#69]
Thanks for your code,I find a bug when i use it.
When I try to  eocode a underlined domain name(Such as:`test_a.abc.com`),I get a ERROR.Although it is lack of standardization,it can be used by some people.And it can work normaly in some occasions.So I think that it shouldn't be a error.What about you?

```python
>>> import idna
>>> idna.encode('abc.com')
b'abc.com'
>>> idna.encode('test_1.abc.com')
Traceback (most recent call last):
  File "D:\Program Files\Python37\lib\site-packages\idna\core.py", line 263, in alabel
    ulabel(label)
  File "D:\Program Files\Python37\lib\site-packages\idna\core.py", line 299, in ulabel
    check_label(label)
  File "D:\Program Files\Python37\lib\site-packages\idna\core.py", line 253, in check_label
    raise InvalidCodepoint('Codepoint {0} at position {1} of {2} not allowed'.format(_unot(cp_value)
, pos+1, repr(label)))
idna.core.InvalidCodepoint: Codepoint U+005F at position 5 of 'test_1' not allowed

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "D:\Program Files\Python37\lib\site-packages\idna\core.py", line 355, in encode
    result.append(alabel(label))
  File "D:\Program Files\Python37\lib\site-packages\idna\core.py", line 265, in alabel
    raise IDNAError('The label {0} is not a valid A-label'.format(label))
idna.core.IDNAError: The label b'test_1' is not a valid A-label

**Comment by jrlevine** [comment]
That's how it's supposed to work. IDNA2008 specifically lists the characters allowed in hostnames, and _ is not one of them.
It would be a good idea to have some options to deal with domain names that are not hostnames. In your case an option to allow ASCII names with non-hostname characters. Another options could do NFC or UTS46 preprocessing.
FYI, although it is valid to have an underscore in a domain name, it's not valid in a hostname that names a computer such as a web server.

**Comment by ZhaoQi99** [comment]
Thanks,I think is really a good idea to add an option to solve this problem.
FYI, it could add an option such as `errors="weak"` which can deal with domain names that are not hostnames.

**Comment by luto** [comment]
@ZhaoQi99 FYI: this has been reported (and closed) three times already. It struck to me, too, today, which is why I am even here ;) There is a work around available in https://github.com/kjd/idna/issues/50#issuecomment-449699205.

**Comment by ZhaoQi99** [comment]
> @ZhaoQi99 FYI: this has been reported (and closed) three times already. It struck to me, too, today, which is why I am even here ;) There is a work around available in [#50 (comment)](https://github.com/kjd/idna/issues/50#issuecomment-449699205).



> @ZhaoQi99 FYI: this has been reported (and closed) three times already. It struck to me, too, today, which is why I am even here ;) There is a work around available in [#50 (comment)](https://github.com/kjd/idna/issues/50#issuecomment-449699205).

Thanks!
